multidim_threshold/search.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import multidim_threshold
+from collections import namedtuple
+import more_itertools
+
+project_along = namedtuple('project_along', 'root direc')
+
+
+new_hi = lambda pi, hi: pi.root + np.multiply(pi.direc, np.true_divide(np.array(hi) - np.array(pi.root), np.array(pi.direc)).min())
+midpoint = lambda a, b: np.true_divide(np.array(a) + np.array(b), 2)
+
+def project_along_axes(lo, mid):
+    return [project_along_i(lo[:], mid, i) for i in range(len(lo))]
+
+def project_along_i(lo, mid, i):
+    lo[i] = mid[i]
+    return lo
+
+
+def project_singleLambda(hi, is_member, pi):
+    new_high = new_hi(pi, hi)
+    rec = multidim_threshold.to_rec(pi.root, new_high)
+    map = {}
+    for i in range(len(is_member)):
+        bool_oracle = isinstance(is_member[i](rec.bot), bool)
+        diagsearch = multidim_threshold.binsearch if bool_oracle else multidim_threshold.weightedbinsearch
```

Comment:
I think we should let `diagsearch` be put as an explicit keyword argument:
new signature:

`def project_singleLambda(hi, is_member, pi, *, diagsearch=None):`

then if it's not `None` do this check. Otherwise, I'm perfectly happy overwriting the default (say with an optimization procedure).